salesforce/read_test.go

```diff
@@ -0,0 +1,214 @@
+package salesforce
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/interpreter"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/amp-labs/connectors/test/utils/mockutils"
+	"github.com/go-test/deep"
+)
+
+func TestRead(t *testing.T) { //nolint:funlen,gocognit,cyclop,maintidx
+	t.Parallel()
+
+	responseUnknownObject := mockutils.DataFromFile(t, "unknown-object.json")
+	responseLeadsFirstPage := mockutils.DataFromFile(t, "read-list-leads.json")
+	responseListContacts := mockutils.DataFromFile(t, "read-list-contacts.json")
+
+	tests := []struct {
+		name         string
+		input        common.ReadParams
+		server       *httptest.Server
+		connector    Connector
+		comparator   func(serverURL string, actual, expected *common.ReadResult) bool // custom comparison
+		expected     *common.ReadResult
+		expectedErrs []error
+	}{
+		{
+			name: "At least one field must be provided",
+			server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
to make this test more readable, `input` should explicitly written out

Reply:
This happened before. I will be creating test refactoring PR. As a chore I will apply this to all tests then.